internal/config_loader/adapter_config_types.go

```
@@ -0,0 +1,58 @@
+// Contains infrastructure settings that can be overridden via environment variables
+// and CLI flags using Viper.
+type AdapterConfig struct {
+	APIVersion string            `yaml:"apiVersion" mapstructure:"apiVersion" validate:"required"`
+	Kind       string            `yaml:"kind" mapstructure:"kind" validate:"required,eq=AdapterConfig"`
+	Metadata   Metadata          `yaml:"metadata" mapstructure:"metadata"`
+	Spec       AdapterConfigSpec `yaml:"spec" mapstructure:"spec"`
+}
+
+// AdapterConfigSpec contains the deployment specification
+type AdapterConfigSpec struct {
+	Adapter   AdapterInfo   `yaml:"adapter" mapstructure:"adapter"`
```

golangci-lint (2.5.0)

[error] File is not properly formatted

(gofmt)